syntax = "proto3";

package bookstore;

// Definição do serviço de Autenticação de Usuários
service AuthService {
    rpc Register (RegisterRequest) returns (AuthResponse);
    rpc Login (LoginRequest) returns (AuthResponse);
}

// Mensagens utilizadas pelo serviço
message RegisterRequest {
    string username = 1;
    string password = 2;
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message AuthResponse {
    string token = 1;
}
